### PR TITLE
Added tests for notifications

### PR DESCRIPTION
**Pull request by sbryzak**
Added a few tests to ensure that all notification types created correctly populate the user email property.

Fixes https://issues.redhat.com/projects/CRT/issues/CRT-1221?filter=myopenissues

### CODE REVIEW

File: test/e2e/usersignup_test.go [hunk 1]
```diff
@@ -136,6 +136,33 @@ func (s *userSignupIntegrationTest) TestAutomaticApproval() {
...
+	// when
+	userSignup, _ := NewSignupRequest(s.T(), s.Awaitilities).
+		Username("bartsimpson").
+		Email("bartsimpson@redhat.com").
+		RequireConditions(ConditionSet(Default(), ApprovedAutomatically())...).
+		Execute().Resources()
+
+	// then
+	userSignup, err := s.Host().WaitForUserSignup(userSignup.Name,
+		wait.UntilUserSignupHasConditions(ConditionSet(Default(), ApprovedAutomatically())...),
```
Comment (MatousJobanek) [inline test/e2e/usersignup_test.go:152]: isn't the presence of the conditions already ensured by `RequireConditions(ConditionSet(Default(), ApprovedAutomatically())...).` ?

File: test/e2e/user_management_test.go [hunk 2]
```diff
@@ -314,10 +345,19 @@ func (s *userManagementTestSuite) TestUserDeactivation() {
 			murMember1.Status.ProvisionedTime.String())
 
 		// The user should be set to deactivating, but not deactivated
-		_, err = hostAwait.WaitForUserSignup(userSignupMember1.Name, wait.UntilUserSignupHasConditions(
+		userSignupMember1, err = hostAwait.WaitForUserSignup(userSignupMember1.Name, wait.UntilUserSignupHasConditions(
 			ConditionSet(Default(), ApprovedByAdmin(), Deactivating())...))
 		require.NoError(s.T(), err)
 
+		// Verify that the deactivating notification was created
+		notifications, err := hostAwait.WaitForNotifications(userSignupMember1.Name, toolchainv1alpha1.NotificationTypeDeactivating, 1)
```
Comment (MatousJobanek) [inline test/e2e/user_management_test.go:353]: IIRC, we create notifications using the "compliant username" and not usersignup name.

File: test/e2e/usersignup_test.go [hunk 3]
```diff
@@ -136,6 +136,33 @@ func (s *userSignupIntegrationTest) TestAutomaticApproval() {
...
+		wait.UntilUserSignupHasStateLabel(toolchainv1alpha1.UserSignupStateLabelValueApproved))
+	require.NoError(s.T(), err)
+
+	notifications, err := s.Host().WaitForNotifications(userSignup.Name, toolchainv1alpha1.NotificationTypeProvisioned, 1)
```
Comment (MatousJobanek) [inline test/e2e/usersignup_test.go:156]: the same - you should use compliant username instead

File: test/e2e/usersignup_test.go [hunk 4]
```diff
@@ -136,6 +136,33 @@ func (s *userSignupIntegrationTest) TestAutomaticApproval() {
...
+	require.NoError(s.T(), err)
+
+	VerifyResourcesProvisionedForSignup(s.T(), s.Awaitilities, userSignup, "deactivate30", "base")
```
Comment (MatousJobanek) [inline test/e2e/usersignup_test.go:159]: we don't need to verify the created resources - let's focus only on the notification.

**Review by MatousJobanek — COMMENTED** [review]
I think that for the purpose of the Jira issue:
> Update e2e tests to verify that recipient is always set in Notification CRs

it should be sufficient (and maybe even better) to move the assertion
```
require.NotEmpty(t, notifications.Spec.Recipient)
require.Equal(t,<recipient>, notifications.Spec.Recipient)
```
into 
```
hostAwait.WaitForNotifications
``` 
function so it makes sure that the recipient field is always correctly set,  no matter what kind of notification it is.

**Comment by alexeykazakov** [comment]
This PR is almost one year old. Please reopen it if/when you still want to work on it.

**Pull request closed without merging**